### DISCUSSION TITLE
Jellyfin QNAP backup - How to?

### DISCUSSION THREAD

**Discussion by Oednerich** [pdulvp/jellyfin-qnap#80] [General]
Hi there,
I hope this is the right place to ask this! Before I forget, many thanks to pdulvp for giving users like me an easy way to use jellyfin on their NAS, I love it.
Before upgrading from 10.8.13 to jellyfin 10.9.x on my QNAP TS-664, I wanted to do a configuration backup at least, since the official blog post regarding the update mentioned this and I'm feeling safer that way.
Since I can't just go into the jellyfin folder and backup everything via filestation (or is that possible?), I think I will need to use ssh and backup folders from there, right?
But what folders/files are actually needed? The QNAP OS path structure seems a bit different from what I read on the official jellyfin page.
I'm talking about everything regarding the server configuration and users, I'm not interested in saving playlists or user history or what not.
Can someone point me in the right direction?

thanks and kind regards

edit: just saw in the admin interfaces upper right corner that there are (some, all?) jellyfin paths in the file system.
can I assume that i should have saved everything if i make a copy of "/share/CACHEDEV1_DATA/.qpkg/jellyfin" ?

**Comment by pdulvp** [comment]
Hi, 
Yes to do a backup, you can connect to ssh to the NAS and backup folders under /share/CACHEDEV1_DATA/.qpkg/jellyfin : 
cache/  
conf/  
database/ 
logs/

The other folders (jellyfin/jellyfin-web and jellyfin-ffmpeg shall not have been modified and are the ones that are replaced when updating).

**Reply by Oednerich** [reply]
Hey pdulvp,
thank you very much for clarifying!
While starting the copy process I wondered if it's a bad idea doing this while the jellyfin server is running.
Is there a way to stop the jellyfin app from booting or running via QTS or should I not bother about that?

edit: ok, sorry for bothering, I just found out that you can stop apps via the app center, I will do that

**Reply by ahasenfr** [reply]
> Hi, Yes to do a backup, you can connect to ssh to the NAS and backup folders under /share/CACHEDEV1_DATA/.qpkg/jellyfin : cache/ conf/ database/ logs/

I tried that and even stopped jellyfin before.
But I still get an error while downloading the files:

![image](https://github.com/user-attachments/assets/7fb3fb49-13ed-495d-a12f-3fcef9733b90)

Even though I'm in with admin rights...

**Reply by luxychris2009** [reply]
> Hi, Yes to do a backup, you can connect to ssh to the NAS and backup folders under /share/CACHEDEV1_DATA/.qpkg/jellyfin : cache/ conf/ database/ logs/
> 
> The other folders (jellyfin/jellyfin-web and jellyfin-ffmpeg shall not have been modified and are the ones that are replaced when updating).

@pdulvp hello, my friend!  i would LOVE to be able to backup my jellyfin application on my qnap.  i have a brand new qnap running hero (zfs).  i do not have caching enabled.  this data folder that you shared in your example seems to only be a temporary folder for caching.  IS THERE ANOTHER FOLDER THAT I CAN BACKUP?  perhaps a more permanent folder structure?  thank you for any help!!  blessings, chris

**Reply by luxychris2009** [reply]
okay, just in case it helps someone else, this is where mine is located (i am assuming it changes with every QNAP).  /share/ZFS530_DATA/.qpkg/jellyfin/ ... note that i do not use caching on my system.  :)

**Comment by kq76** [comment]
Sorry, is ssh necessary? Can you not just back up the folders via File Station? 

I screwed up my update and am trying to learn what needs to be done next time.

**Reply by Oednerich** [reply]
hey,
you cannot use File Station as it cannot access the qnap OS folders where jellyfin (or other installed qnap packages) is located. It's only there to create and manage network shares on your volumes.
using ssh for backup is not complicated, if you need some pointers on the how-to just ask

edit: or you can take a look an the official qnap FAQ instructions: 
https://www.qnap.com/en/how-to/faq/article/how-to-access-qnap-nas-file-system-by-winscp

**Reply by kq76** [reply]
I'm confident I can SSH now, but could you take a look at [this thread](https://github.com/pdulvp/jellyfin-qnap/discussions/101) and maybe give me those pointers you mentioned? Thanks.

**Reply by pixelstuff** [reply]
Is there any reason not to create a shared folder (call it JellyfinData or whatever) and point it to the /.qnap/jellyfin folder?

**Comment by pixelstuff** [comment]
I'm not sure if there would be any problems with this, but what about creating a shared folder, call it JellyfinData or whatever, and point it to the /.qpk/jellyfin/ folder.
